### ISSUE TITLE
Explosion knockback in wrong direction

### ISSUE THREAD

**Issue by MaxedOut1** [TitaniumMC/Titanium#137]
Explosion knockback pushes players in the wrong direction. I'm using the latest titanium Dev build, however the issue is also present in the latest public release.

I have currently only tested this with explosions made by plugins, for example the /fireball command from essentialsx and the world.createexplosion() method I used in my own plugin.

If the player is in range of these explosions they are pulled towards it instead of pushed away.

It's possible it's an incompatibility with a plugin i have on my server, however it worked just fine before I started using titanium.

**Comment by P3ridot** [comment]
Do you have async tnt enabled?

**Comment by MaxedOut1** [comment]
[titanium.yml.txt](https://github.com/TitaniumMC/Titanium/files/10679842/titanium.yml.txt)
Heres my config. (as txt cause thats what github supports.)

Async tnt is currently disabled, but I will test whether enabling it fixes this problem

**Comment by P3ridot** [comment]
Configuration looks normal, could you please create videos showing difference between Titanium and plain Paper?

**Comment by MaxedOut1** [comment]
Just tested with async enabled, issue is still there. I will test with regular paper now. None of my plugins should be changing velocity or knockback, but you never know. Will report back shortly

**Comment by MaxedOut1** [comment]
Ok just tested with regular paper, explosion knockback is completely normal. The issue is only there when using titanium.
No errors show up in console with paper or with titanium.

So far I have only tested with explosions made by plugins (essentialsx and my own), however I will switch back to titanium and test with regular tnt explosions to find out if its something to do with plugin generated explosions.

The issue is exactly as I described, with titanium the player is pulled towards the explosion, rather with paper they are pushed away.

**Comment by P3ridot** [comment]
Right, I'll try to test and fix this when I would have some time. It's unfortunately really strange issue and currently I have no idea what is causing that...

**Comment by MaxedOut1** [comment]
Tested again with titanium and regular tnt, issue is still there. Good luck and let me know if you need any more testing!

**Comment by P3ridot** [comment]
I noticed that tnt on titanium works a little different than on standard paper. I'd have to do some more testing ans debugging to actually find the reasons it's happening
